### DISCUSSION TITLE
Hihat circuits

### DISCUSSION THREAD

**Discussion by jstma** [corrados/edrumulus#65] [General]
I'm working on a circuit to attach a Roland FD-9 to a Yamaha DTX M12 which is not related to this project, but I have a couple of yamaha pedal circuits to share.

The first one is an HH65 pedal:

![VHYHHCHiHatSchematic](https://user-images.githubusercontent.com/87273793/188193492-b2bb0ed1-c786-435b-9b84-db3cd4ab17b5.jpg)

Originally from [here](https://www.toontrack.com/forums/topic/design-hihat-controller-for-yamahatm-trigger-modules-by-vanhouten/).

The second one is an RHH135:

![rhh135-schematic](https://user-images.githubusercontent.com/87273793/188194051-fc71f2b1-d136-4252-9d2c-d16f5962185f.png)

That I drew up in kicad based on [here](https://www.vdrums.com/forum/general/products/1130882-vh-11-and-td-9-hi-hat-positions-resistance-values) and [here](https://www.vdrums.com/forum/general/products/1133932-guide-converting-a-yamaha-rhh135-to-function-like-a-roland-vh-11).

What's interesting about these Yamaha designs is the added capacitor.  It adds a slope to changes in the switches (or pedal position).  I presume it's there for foot splash detection?

The FD-9 pedal I have doesn't have any parts on it other than the FSR which measures approx 20K with no pressure on it.

Also the DTX M12 has an approximate 110K resistor in series with the tip inside the module.  I believe it's there to act as a current limiter.  This doesn't really matter when interfacing a Yamah pedal with edrumulus, but I thought I'd mention it.

**Comment by corrados** [comment]
Yamaha is working with switches? That it an interesting concept. Maybe they match the number of switches to the number of different samples they have for the hi-hat.

>  I presume it's there for foot splash detection?

Maybe the have a smooth transmission between the "hard switch voltage levels"? Maybe some Yamaha units just use the switches and some better units use the additional information about the smoothed voltage level. But actually, you could do this in software as well :-).

> The FD-9 pedal I have doesn't have any parts on it other than the FSR which measures approx 20K with no pressure on it.

I also [measured the hi-hat pedals I own](https://github.com/corrados/edrumulus/blob/main/doc/ChangeLog.md#2021-03-24-measurements-of-hi-hat-controllers).

**Reply by jstma** [reply]
If I had to guess Roland must have patented FSRs or potentiometers so Yamaha had to go a different route and use what is essentially a resistor ladder.  As far as I know, no Yamaha pedals use FSRs.

The switches are not physical switches.  There is a membrane that looks kinda like an FSR that has different switches in them.  When you press down on the pedal this rubbery piece with an arc in it presses down on that membrane in a progressive manner connecting across one switch, then two, then three, etc.

I can't see why "hard switch voltage levels" would matter to the module.  According to my spice model a transition from 3.3V to 0V (unpressed, fully pressed) happens with a sharp edge.  That capacitor only comes into play when the pedal rises.

Hopefully edrumulus can support all the drum equipment out there, even hihat pedals from Yamaha ;p
